crates/solvers/src/infra/config.rs

```diff
@@ -52,6 +52,7 @@ struct Config {
 
     /// If this is configured the solver will also use the Uniswap V3 liquidity
     /// sources that rely on RPC request.
+    #[serde(default, deserialize_with = "deserialize_url")]
```

Comment:
Instead of writing your own deserialization logic you can add `url = { workspace = true, features = ["serde"] }` to `solvers/Cargo.toml`.